apps/sms/views/shared/js/startup.js

```diff
@@ -89,14 +89,15 @@ var Startup = exports.Startup = {
 
       // Init UI Managers
       TimeHeaders.init();
-      ConversationView.init();
-      MessagingClient.init(App.instanceId);
       MozMobileConnectionsClient.init(App.instanceId);
-      Information.initDefaultViews();
+      MessagingClient.init(App.instanceId).then(() => {
```

Comment:
I think I mess it with my previous implementation that assign the client's on method to service's on method and it will need to wait for the service is actually ready. So yes you're right that client side doesn't need to wait for the service side event registration ready with current implementation.